features/pakiti/client/config.pan

```diff
@@ -6,34 +6,87 @@ include PAKITI_RPMS;
+variable PAKITI_SERVER_HTTP_FEED_URL ?= '/feed/';
+variable PAKITI_SERVER_PUB_KEY_FILE ?= '/etc/pakiti/client/server-pub-key.pem';
+variable PAKITI_SERVER_PUB_KEY_SOURCE ?= 'features/pakiti/client/server-pub-key.pem';
+variable PAKITI_SERVER_CERTIFICATION_CHAIN_FILE ?= '/etc/pakiti/pakiti-server-ca.pem';
+# PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT: see below how it is used
+variable PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT ?= 'features/pakiti/client/server-ca.pem';
+variable PAKITI_CLIENT_SLEEP ?= 7200;
 variable PAKITI_CLIENT_UPDATE_FREQ ?= "45 13 * * *";
 
-variable pakiti_conf = format("site = %s\n", PAKITI_TAG);
-variable pakiti_conf = pakiti_conf + format("url = http://%s:%s%s\n",
-                                            PAKITI_SERVER,
-                                            PAKITI_SERVER_HTTP_PORT,
-                                            PAKITI_SERVER_HTTP_FEED_URL);
-variable pakiti_conf = pakiti_conf + "expect = 200 OK\n";
-variable pakiti_conf = pakiti_conf + "encrypt = <<EOT\n" + PAKITI_SERVER_PUB_KEY + "EOT\n";
-variable pakiti_conf = pakiti_conf + format("rndsleep = %s\n", PAKITI_CLIENT_SLEEP);
+# If PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE is not explicitely defined,
+# its value is PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT is the file exists
+# else it remains undefined and PAKITI_SERVER_CERTIFICATION_CHAIN_FILE is not created
+# If PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE is defined, the file referred to must exist.
+variable PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE ?= {
+    if ( file_exists(PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT) ) {
+        PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT;
+    } else {
+        undef;
+    };
+};
 
+# Build configuration file
+include 'components/metaconfig/config';
+include 'features/pakiti/client/config_schema';
+prefix '/software/components/metaconfig/services/{/etc/pakiti/pakiti-client.conf}';
+'backup' = '.old';
+'module' = 'tiny';
+'contents' = {
+    SELF['site'] = PAKITI_TAG;
+    SELF['url'] = format(
+        "%s://%s:%s%s",
+        PAKITI_SERVER_PROTOCOL,
+        PAKITI_SERVER,
+        PAKITI_SERVER_HTTP_PORT,
+        PAKITI_SERVER_HTTP_FEED_URL,
+    );
+    if ( is_defined(PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE) ) {
+        SELF['curl'] = format("curl --cacert %s", PAKITI_SERVER_CERTIFICATION_CHAIN_FILE);
+    };
+    # If PAKITI_SERVER_PUB_KEY_SOURCE is an existing file, define the encrypt
+    # parameter. Else raise an error if the protocol is not https.
+    if ( file_exists(PAKITI_SERVER_PUB_KEY_SOURCE) ) {
+        SELF['encrypt'] = PAKITI_SERVER_PUB_KEY_FILE;
+    } else if ( PAKITI_SERVER_PROTOCOL != 'https' ) {
+        error('Pakiti server pub key is missing and feed protocol is not https');
+    };
+    SELF['rndsleep'] = PAKITI_CLIENT_SLEEP;
+    SELF;
+};
+'convert/yesno' = true;
+bind '/software/components/metaconfig/services/{/etc/pakiti/pakiti-client.conf}/contents' = pakiti_client_config;
```

Comment:
If this is for Pakiti v3 can we make this:
```suggestion
bind '/software/components/metaconfig/services/{/etc/pakiti3/pakiti-client.conf}/contents' = pakiti_client_config;
```

Reply:
Done